roles/openshift_certificate_expiry/library/openshift_cert_expiry.py

```diff
@@ -268,30 +270,23 @@ def load_and_handle_cert(cert_string, now, base64decode=False, ans_module=None):
     # around a missing library on the target host.
     #
     # pylint: disable=redefined-variable-type
-    if 'OpenSSL.crypto' in sys.modules:
+    if HAS_OPENSSL:
         # No work-around required
         cert_loaded = OpenSSL.crypto.load_certificate(
             OpenSSL.crypto.FILETYPE_PEM, _cert_string)
     else:
-        # Missing library, work-around required. We need to write the
-        # cert out to disk temporarily so we can run the 'openssl'
+        # Missing library, work-around required. Run the 'openssl'
         # command on it to decode it
-        _, path = tempfile.mkstemp()
-        with io.open(path, 'w') as fp:
-            fp.write(six.u(_cert_string))
-            fp.flush()
-
-        cmd = 'openssl x509 -in {} -text'.format(path)
+        cmd = 'openssl x509 -text'
         try:
-            openssl_decoded = subprocess.Popen(cmd.split(),
-                                               stdout=subprocess.PIPE)
+            openssl_proc = subprocess.Popen(cmd.split(),
```

Comment:
After some digging, `call`, `check_call`, and `check_output` all use `communicate` internally, so the alternative to `Popen` and `communicate` would involve pre-creating a `PIPE` and writing to that `PIPE` while `check_output` is blocking on a result.

I thought that it might be possible to handle it through `StringIO`, but since it doesn't produce a `fileno` it cannot be used with subprocess.

Python 3.4 introduced the `input` argument to `check_output`, which would solve the issue, but not for Python 2.7 :(